octoprint_mrbeam/__init__.py

```diff
@@ -1748,162 +1748,174 @@ def gcodeConvertCommand(self):
+                        slicer,
+                        FileDestinations.LOCAL,
+                        filename,
+                        FileDestinations.LOCAL,
                         gcode_name,
-                        select_after_slicing,
-                        print_after_slicing,
-                        appendGcodeFiles,
-                    ],
-                )
-            except octoprint.slicing.UnknownProfile:
-                msg = "Profile {profile} doesn't exist".format(**locals())
-                self._logger.error("gcodeConvertCommand: %s", msg)
-                return make_response(msg, 400)
-
-            location = "test"  # url_for(".readGcodeFile", target=target, filename=gcode_name, _external=True)
-            result = {
-                "name": gcode_name,
-                "origin": "local",
-                "refs": {
-                    "resource": location,
-                    "download": url_for("index", _external=True)
-                    + "downloads/files/"
-                    + FileDestinations.LOCAL
-                    + "/"
-                    + gcode_name,
-                },
-            }
+                        profile=None,  # profile,
+                        printer_profile_id=None,  # printerProfile,
+                        position=None,  # position,
+                        overrides=overrides,
+                        callback=slicing_done,
+                        callback_args=[
+                            gcode_name,
+                            select_after_slicing,
+                            print_after_slicing,
+                            appendGcodeFiles,
+                        ],
+                    )
+                except octoprint.slicing.UnknownProfile:
+                    msg = "Profile {profile} doesn't exist".format(**locals())
+                    self._logger.error("gcodeConvertCommand: %s", msg)
+                    return make_response(msg, 400)
+
+                location = "test"  # url_for(".readGcodeFile", target=target, filename=gcode_name, _external=True)
+                result = {
+                    "name": gcode_name,
+                    "origin": "local",
+                    "refs": {
+                        "resource": location,
+                        "download": url_for("index", _external=True)
+                        + "downloads/files/"
+                        + FileDestinations.LOCAL
+                        + "/"
+                        + gcode_name,
+                    },
+                }
 
-            r = make_response(jsonify(result), 202)
-            r.headers["Location"] = location
-            return r
+                r = make_response(jsonify(result), 202)
+                r.headers["Location"] = location
+                return r
+            except Exception as e:
+                self._logger.exception('Exception in gcodeConvertCommand() "/convert":')
+                raise e
```

Comment:
Use the `@logExceptions` decorator instead. It also avoids lengthy diffs because of indentation and logs the function name.